Update migration docs to pass in AWS creds to stork

disrani-px opened the pull request.

disrani-px committed:
Update migration docs to pass in AWS creds to stork

wallnerryan commented on cloud/migration-stork.md:
```diff
@@ -115,14 +115,59 @@ status:
+        args:
+        - token
+        - -i
+        - demo-ryan-cluster2
```
may want to change this to something more generic. "my-cluster" or "demo-cluster"

disrani-px replied:
whoops, yeah I meant to do that.

wallnerryan commented on cloud/migration-stork.md:
```diff
@@ -115,14 +115,59 @@ status:
-5. Once you apply the above spec you should be able to check the status of the pairing. On a successful pairing, you should 
+5. (EKS Only) When pairing with an EKS cluster, you also need to pass in your
+   AWS credentials which will be used to generate the IAM token. This can be
+   achieved in 2 ways
```
when first reading this, I thought it was both, maybe we can add a simple warning of "choose only one" or OPTIONAL next to `.2` or something.

disrani-px replied:
Updated, let me know if that sounds fine.

disrani-px committed:
Update migration-stork.md

wallnerryan commented on cloud/migration-stork.md:
```diff
@@ -115,14 +115,59 @@ status:
+5. (EKS Only) When pairing with an EKS cluster, you also need to pass in your
+   AWS credentials which will be used to generate the IAM token. This can be
+   achieved in 2 ways
+   1. Create a secret and mount in the stork deployment (Secure)
```
It wasn't clear that this was supposed to happen on the source cluster. Lets try to be overly obvious about that for readers.

disrani-px replied:
Done.

wallnerryan commented on cloud/migration-stork.md:
````diff
@@ -115,14 +115,59 @@ status:
+       $ kubectl create secret  generic --from-file=$HOME/.aws/credentials -n  kube-system aws-creds
+       secret/aws-creds created
+       ```
+       2. Mount the secret created above in the stork deployment. Run `kubectl edit deployment -n kube-system stork` and make the following updates.
````
maybe add the command `kubectl get po -n kube-system -l name=stork` after this to tell users to make sure stork was updated properly

disrani-px replied:
Added

wallnerryan commented on cloud/migration-stork.md:
````diff
@@ -115,14 +115,59 @@ status:
 ```
 $ storkctl get clusterpair
 NAME            STORAGE-STATUS   SCHEDULER-STATUS   CREATED
 remotepair      Ready            Ready              26 Oct 18 03:11 UTC
````
somewhere, idk where but maybe worth adding FAQ or common issues? Or at least mention that you should make sure 9001 and 9010 should be open in firewalls. my CF template doesnt open them even within a VPC and i just have to open them not sure if its in the doc anywhere else.

disrani-px replied:
Added to the pre-reqs at the top of the page

wallnerryan reviewed (commented):
Hey @disrani-px thanks for this. I made a few suggestions but verified this works for EKS --> EKS now.

disrani-px committed:
Update with review comments

disrani-px committed:
Add instructions to generate cluster pair for GKE

Also fix yaml formatting which shows up fine in github but wasn't aligned
in jekyll

disrani-px committed:
Update with review comments

wallnerryan commented:
@disrani-px LGTM. The only things i would add would be 
- a better structure for GKE, EKS, AKS ...etc but that should be a different issue/PR.
- possibly once its tested enough, add the "extra" bits to stork deployment via http://install.portworx.com instead of editing the deployment (this also is a separate issue/PR)

wallnerryan commented on cloud/migration-stork.md:
````diff
@@ -115,14 +117,105 @@ status:
+   1. Create a service account key using [the guide from Google Cloud](https://cloud.google.com/iam/docs/creating-managing-service-account-keys)
+   and save it as gcs-key.json. You can also create this using the following command: 
+   ```
+   $ gcloud iam service-accounts keys create gcs-key.json \
````
One thing I ran into while continuing to use this and test it is I have to add permissions for this service accound to the *destination cluster* that allow it to create things like namespaces etc. It might be a good idea to add a note or an example of this such as

Where `<cluster-role>` has access to create resources, ive used `cluster-admin` for this but their may be more appropriate roles

```
$ kubectl create clusterrolebinding <svc-accnt> --clusterrole=<cluster-role> --user=<svc-accnt>@<project>.iam.gserviceaccount.com
```

disrani-px replied:
Meeting this PR for now, will make further changes in a separate PR
